src/OrchardCore.Modules/OrchardCore.Rules/Drivers/ContentTypeConditionEvaluatorDriver.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Display.ContentDisplay;
+using OrchardCore.ContentManagement.Metadata.Settings;
+using OrchardCore.DisplayManagement.Handlers;
+using OrchardCore.DisplayManagement.Views;
+using OrchardCore.Rules.Models;
+
+namespace OrchardCore.Rules.Drivers
+{
+    /// <summary>
+    /// Saves references to content types which have been displayed during a request.
+    /// </summary>
+    public class ContentTypeConditionEvaluatorDriver : ContentDisplayDriver, IConditionEvaluator
+    {
+        private static ValueTask<bool> True = new ValueTask<bool>(true);
+        private static ValueTask<bool> False = new ValueTask<bool>(false);
+
+        private readonly IConditionOperatorResolver _operatorResolver;
+
+        // Hashset to prevent duplicate entries, but comparison is done by the comparers.
+        private readonly HashSet<string> _contentTypes = new HashSet<string>();
+
+        public ContentTypeConditionEvaluatorDriver(IConditionOperatorResolver operatorResolver)
+        {
+            _operatorResolver = operatorResolver;
+        }
+
+        public override Task<IDisplayResult> DisplayAsync(ContentItem contentItem, BuildDisplayContext context)
+        {
+            // Do not include Widgets or any display type other than detail.
+            if (context.DisplayType == "Detail" && 
+                (context.Shape.Properties.TryGetValue(nameof(ContentTypeSettings.Stereotype), out var stereotype) && 
+                stereotype == null))
+            {
+                _contentTypes.Add(contentItem.ContentType);
+            }
+
+            return Task.FromResult<IDisplayResult>(null);
+        }
+
+        ValueTask<bool> IConditionEvaluator.EvaluateAsync(Condition condition)
```

Comment:
Is it necessary to implement the interface explicitly? As i remember i removed some in Orchard that was not necessary, maybe i also needed to update the services registrations, but not sure

Reply:
Yes, in this case, because the driver inherits from `ContentDisplayDriver` but is also an `IConditionEvaluator`

implementing interface explicitly, in a similar way in which we do the `IContentPartDisplayDriver` - to do the dispatch to a `TPart`, or in this case a `TCondition`.

Maybe there is a better pattern for dispatch? 
but I've learnt that one since O1 as well ;)

Reply:
Okay no problem, and maybe better for readability, and it is required when a class implement 2 methods with the same signature but for 2 different interfaces, but not the case here, so just to say that here we could also use

        public ValueTask<bool> EvaluateAsync(Condition condition)
            => EvaluateAsync(condition as ContentTypeCondition);

Reply:
Ok, cool. was there a disadvantage to it? 
I quite liked it for the readability... but don't mind if there's a preference the other way.

Just noticed the `Evaluate` can just be private, so changing that.

Reply:
@deanmarcussen 

No disadvantage, was just curious if there was any other reason, yes about readability

Reply:
@jtkech looked again, and yes it was a bit odd on that driver :) cheers for spotting.

I've kept it on `ConditionEvaluator<T>` as it looks more obvious to me when I read it.

Any other thing you saw? will probably merge it tomorrow if not.

Reply:
Nothing else, i didn't have time to review it more, but i'm confident ;)